Drivers/NER/include/cascadiamc.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file cascadiamc.h
+ * @author Hamza Iqbal
+ * @brief Driver to abstract sending and receiving CAN messages to control cascadia motor controller
+ * @version 0.1
+ * @date 2023-08-09
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+
+#ifndef CASCADIAMC_H
+#define CASCADIAMC_H
+
+#include <stdint.h>
+#include "stm32f4xx_hal.h"
+#include "can.h"
+
+#define NUM_POLES
+typedef struct
+{
+    int32_t rpm;            /* SCALE: 1         UNITS: Rotations per Minute   */
+    int16_t duty_cycle;     /* SCALE: 10        UNITS: Percentage             */
+    int16_t input_voltage;  /* SCALE: 1         UNITS: Volts                  */
+    int16_t ac_current;     /* SCALE: 10        UNITS: Amps                   */
+    int16_t dc_current;     /* SCALE: 10        UNITS: Amps                   */
+    int16_t cont_temp;      /* SCALE: 10        UNITS: Degrees Celsius        */
+    int16_t motor_temp;     /* SCALE: 10        UNITS: Degrees Celsius        */
+    uint8_t fault_code;     /* SCALE: 1         UNITS: No units just a number */
+    int8_t throttle_signal; /* SCALE: 1         UNITS: Percentage             */
+    int8_t brake_signal;    /* SCALE: 1         UNITS: Percentage             */
+    int8_t drive_enable;    /* SCALE: 1         UNITS: No units just a number */
+} cascadiamc_t;
+
+cascadiamc_t_t* cascadiamc;
+
+void cascadiamc_init(cascadiamc_t* mc);
+
+void cascadiamc_update(cascadiamc_t* mc);
+
+int32_t cascadiamc_get_rpm(cascadiamc_t* mc);
+
+int16_t cascadiamc_get_duty_cycle(cascadiamc_t* mc);
+
+int16_t cascadiamc_get_input_voltage(cascadiamc_t* mc);
```

Comment:
I appreciate the set and get interface, clean way of doing it

Reply:
Actually possibly sike, I like the sets but the getters might not be necessary if we are dealing with a central struct we pass the reference to